src/update_agent/mod.rs

```diff
@@ -57,18 +55,20 @@ impl Default for UpdateAgentState {
 
 impl UpdateAgentState {
     /// Return the discriminant for current state
+    #[allow(dead_code)]
     fn discriminant(&self) -> u8 {
```

Comment:
Do we actually need this function? The only place that was using it is now gone.

Reply:
Yes, dropped. I was planning to use it for checking the "from" state on transition, but I realized I cannot do it that way.

Reply:
Self-note: one approach to explore here is to statically enforce allowed transitions via generics, like https://hoverbear.org/blog/rust-state-machine-pattern/#generically-sophistication.